File: dotnet/src/ShoppingCart.API/Coupon/CreateCouponRequest.cs
using ShoppingCart.Business.Campaign;

namespace ShoppingCart.API.Coupon
{
    public class CreateCouponRequest
    {
        public decimal MinimumCartAmount { get; set; }

        public DiscountType Type { get; set; }

        public decimal Rate { get; set; }
    }
}

File: spring-boot/src/main/java/com/erdaldalkiran/cart/api/coupon/CreateCouponRequest.java
package com.erdaldalkiran.cart.api.coupon;

import com.erdaldalkiran.cart.application.discount.OnlyValidDiscountTypes;
import com.erdaldalkiran.cart.domain.discount.DiscountType;
import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import java.math.BigDecimal;
import java.util.UUID;

@Getter
@Setter
public class CreateCouponRequest {
    @NotNull
    @Positive
    private BigDecimal minimumCartAmount;

    @NotNull
    @OnlyValidDiscountTypes
    private DiscountType discountType;

    @NotNull
    @Positive
    private BigDecimal rate;

}
